File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/messages/TankLocalizerInputsMessage.java
package org.firstinspires.ftc.teamcode.messages;

import com.acmerobotics.roadrunner.ftc.PositionVelocityPair;

import java.util.List;

public final class TankLocalizerInputsMessage {
    public long timestamp;
    public PositionVelocityPair[] left;
    public PositionVelocityPair[] right;

    public TankLocalizerInputsMessage(List<PositionVelocityPair> left, List<PositionVelocityPair> right) {
        this.timestamp = System.nanoTime();
        this.left = left.toArray(new PositionVelocityPair[0]);
        this.right = right.toArray(new PositionVelocityPair[0]);
    }
}

File: TeamCode/src/main/main/java/org/firstinspires/ftc/teamcode/drivetrain/messages/TankLocalizerInputsMessage.kt
package org.firstinspires.ftc.teamcode.drivetrain.messages

import com.acmerobotics.roadrunner.ftc.PositionVelocityPair

class TankLocalizerInputsMessage(
    left: MutableList<PositionVelocityPair?>,
    right: MutableList<PositionVelocityPair?>
) {
    var timestamp: Long
    var left: Array<PositionVelocityPair?>?
    var right: Array<PositionVelocityPair?>?

    init {
        this.timestamp = System.nanoTime()
        this.left = left.toTypedArray<PositionVelocityPair?>()
        this.right = right.toTypedArray<PositionVelocityPair?>()
    }
}
